src/Sulu/Bundle/SnippetBundle/DependencyInjection/Compiler/SnippetAreaCompilerPass.php

```diff
@@ -72,23 +75,36 @@ public function process(ContainerBuilder $container)
     }
 
     /**
-     * Get area.
+     * @param array{key: string, title: array<string>} $area
+     * @param array<string> $locales
+     * @param array<string, string> $templateTitles
      *
-     * @return array
+     * @return array{key: string, template: string, title: array<string, string>}
      */
-    private function getArea($template, $area, $locales, $templateTitles)
+    private function getArea(ContainerBuilder $container, string $template, array $area, array $locales, array $templateTitles): array
     {
         $key = $area['key'];
 
         $titles = [];
+        $areaTitles = $area['title'];
 
-        foreach ($locales as $locale) {
-            $title = $templateTitles[$locale] . ' ' . \ucfirst($key);
-            if (isset($area['title'][$locale])) {
-                $title = $area['title'][$locale];
+        // If we only have one title, it is probably a translation key
+        if (1 === \count($areaTitles)) {
```

Comment:
this looks strange. It should get translated if there is no locale defined for it e.g.:

```xml
<meta>
   <title>app.title</title>
</meta>
```

but following should not be translated:

```xml
<meta>
   <title lang="en">Title</title>
</meta>
```

How is this handled for the other xml templates like snippets, pages, articles, ..?